### DISCUSSION TITLE
Move NPM build step to Dockerfile

### DISCUSSION THREAD

**Discussion by justinmayer** [chartbrew/chartbrew#136] [Ideas]
Currently, there is an NPM build step for the client that takes place in the [Docker entrypoint](https://github.com/chartbrew/chartbrew/blob/master/entrypoint.sh#L13). This means that every time the container is (re-)created, there is a lengthy delay of several minutes while waiting for this build process to complete.

In the vast majority of projects I have seen, this build step is located in the [Dockerfile](https://github.com/chartbrew/chartbrew/blob/master/Dockerfile) and occurs during the Docker image build process, making for lightning-fast deploys when compared to building during container instantiation.

Could we consider moving this NPM build step to the [Dockerfile](https://github.com/chartbrew/chartbrew/blob/master/Dockerfile)?

**Comment by razvanilin** [comment]
Oh that's cool, I didn't know that as I don't use Docker myself. It looks like it's just a matter of moving some of the commands in the entrypoint to the Dockerfile. Thanks for flagging this 🫡
